### DISCUSSION TITLE
Problems compiling 1.12.1

### DISCUSSION THREAD

**Discussion by wmoe21** [gerbera/gerbera#2849] [General]
because I get on a fresh raspian os (bullseye) with "apt-get install gebera" the very old version 1.1.0 - I tried to compile 1.12.1 for my own. But seriously I stuck at the libupnp

CMake Error at CMakeLists.txt:405 (message):
  Gerbera failed to find libupnp! at least version 1.14.6 is needed.

I loaded the libupnp-1.14.17 libupnp-1.14.17 version, set it up with ./bootstrap - ./configure - make - make install
see the following under /usr/local/lib

ls -l /usr/local/lib/
total 3288
-rw-r--r-- 1 root root  215700 Aug  7 21:58 libixml.a
-rwxr-xr-x 1 root root     945 Aug  7 21:58 libixml.la
lrwxrwxrwx 1 root root      17 Aug  7 21:58 libixml.so -> libixml.so.11.1.2
lrwxrwxrwx 1 root root      17 Aug  7 21:58 libixml.so.11 -> libixml.so.11.1.2
-rwxr-xr-x 1 root root  154964 Aug  7 21:58 libixml.so.11.1.2
-rw-r--r-- 1 root root 1273646 Aug  7 21:58 libupnp.a
-rwxr-xr-x 1 root root     971 Aug  7 21:58 libupnp.la
lrwxrwxrwx 1 root root      17 Aug  7 21:58 libupnp.so -> libupnp.so.17.1.9
lrwxrwxrwx 1 root root      17 Aug  7 21:58 libupnp.so.17 -> libupnp.so.17.1.9
-rwxr-xr-x 1 root root  851712 Aug  7 21:56 libupnp.so.17.1.8
-rwxr-xr-x 1 root root  851696 Aug  7 21:58 libupnp.so.17.1.9
drwxr-xr-x 2 root root    4096 Aug  7 21:58 pkgconfig
drwxr-xr-x 3 root root    4096 May  3 01:07 python3.9

but still get this 
Gerbera failed to find libupnp! at least version 1.14.6 is needed.
message
can someone please tell me whats the problem?
thanks for your help ;-)

**Comment by whyman** [comment]
Probably you need to uninstall the the system libupnp, I guess cmake is finding that version first

**Comment by wmoe21** [comment]
it seems there is a problem with the libupnp-dev file.
This one I installed with "apt-get install libupnp-dev"
now I removed it, and get a different Error Message
stupid question, where do I get a -dev file which coorelate to my compiled version?

root@raspberrypi:/home/wmoe/build# cmake ../gerbera -DWITH_MAGIC=1 -DWITH_MYSQL=1 -DWITH_CURL=1 -DWITH_JS=1 -DWITH_TAGLIB=1 -DWITH_AVCODEC=1 -DWITH_FFMPEGTHUMBNAILER=1 -DWITH_EXIF=1 -DWITH_LASTFM=0
-- Gerbera Version: v1.12.1-52-g12063963
-- Looking for BSD native UUID
-- Looking for libuuid
-- Looking for libuuid - found
CMake Error at cmake/FindUPNP.cmake:47 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
Call Stack (most recent call first):
  CMakeLists.txt:396 (find_package)


-- Found libupnp: UPNP::UPNP-NOTFOUND (found version )
CMake Error at CMakeLists.txt:421 (message):
  Gerbera requires libupnp with IPv6 support.

**Reply by KarlStraussberger** [reply]
First you need to uninstall all upnp (lib and dev), then run `install-pupnp.sh` from the `scripts` dir.

**Comment by wmoe21** [comment]
this install-pupnp.sh script will do the same, as I did before manually.
download the pupnp-1.14.17 version, ./configure & make & make install
at the end of the Script there is a
Libraries have been installed in:
   /usr/local/lib

now the Date and Timestamp is different to yesterday:
root@raspberrypi:/usr/local/lib# ls -l
total 3300
-rw-r--r-- 1 root root  216340 Aug  8 08:24 libixml.a
-rwxr-xr-x 1 root root     945 Aug  8 08:24 libixml.la
lrwxrwxrwx 1 root root      17 Aug  8 08:24 libixml.so -> libixml.so.11.1.2
lrwxrwxrwx 1 root root      17 Aug  8 08:24 libixml.so.11 -> libixml.so.11.1.2
-rwxr-xr-x 1 root root  155336 Aug  8 08:24 libixml.so.11.1.2
-rw-r--r-- 1 root root 1279082 Aug  8 08:24 libupnp.a
-rwxr-xr-x 1 root root     971 Aug  8 08:24 libupnp.la
lrwxrwxrwx 1 root root      17 Aug  8 08:24 libupnp.so -> libupnp.so.17.1.8
lrwxrwxrwx 1 root root      17 Aug  8 08:24 libupnp.so.17 -> libupnp.so.17.1.9
-rwxr-xr-x 1 root root  854796 Aug  8 08:24 libupnp.so.17.1.8
-rwxr-xr-x 1 root root  851696 Aug  7 21:58 libupnp.so.17.1.9
drwxr-xr-x 2 root root    4096 Aug  8 08:24 pkgconfig
drwxr-xr-x 3 root root    4096 May  3 01:07 python3.9

but the Error is still there
root@raspberrypi:/home/wmoe/build# cmake ../gerbera -DWITH_MAGIC=1 -DWITH_MYSQL=1 -DWITH_CURL=1 -DWITH_JS=1 -DWITH_TAGLIB=1 -DWITH_AVCODEC=1 -DWITH_FFMPEGTHUMBNAILER=1 -DWITH_EXIF=1 -DWITH_LASTFM=0
-- Gerbera Version: v1.12.1-52-g12063963
-- Looking for BSD native UUID
-- Looking for libuuid
-- Looking for libuuid - found
CMake Error at cmake/FindUPNP.cmake:47 (set_target_properties):
  set_target_properties called with incorrect number of arguments.
Call Stack (most recent call first):
  CMakeLists.txt:396 (find_package)


-- Found libupnp: UPNP::UPNP-NOTFOUND (found version )
CMake Error at CMakeLists.txt:421 (message):
  Gerbera requires libupnp with IPv6 support.


-- Configuring incomplete, errors occurred!
See also "/home/wmoe/build/CMakeFiles/CMakeOutput.log".
See also "/home/wmoe/build/CMakeFiles/CMakeError.log".

**Reply by KarlStraussberger** [reply]
You should remove all the cached files from cmake `CMake*` and also check `/usr/(local/)?(lib|share)/pkgconfig` for  old versions of `pupnp.pc` or so.

**Comment by wmoe21** [comment]
Thanks, it took a while but I got it compiled on a Raspberry Pi 1 ;-)
